core/src/staked_nodes_updater_service.rs

```diff
@@ -21,18 +22,40 @@ pub struct StakedNodesUpdaterService {
     thread_hdl: JoinHandle<()>,
 }
 
+#[derive(Default, Deserialize, Clone)]
+pub struct StakedNodesOverrides {
```

Comment:
I don't love how the placement of this struct here causes a number of extra downstream `use` of the `staked_nodes_updater_service` module, which really feels like it ought to be private to `solana-core`. Likewise, it necessitates a new `serde_yaml` dep for core.

Any way we can move the `serde_yaml` dep fully into `validator/` instead?

Also are these files going to be large? It's been my experience that `serde_yaml` is very slow on large files (relative to `serde_json`)

Reply:
The declaration was used to easily load various data from the struct. It was meant to be similar as other structures coming from `core` to other modules. As it's the `validator` module that needs to provide data to the `core service` I haven't thought an easy way to do so.

The struct could be defined without the `serde_yaml` dependency but then it still means multiple extra downstream `use`.
I haven't thought anything super clever and I eventually pass the `HashMap` as the shared structure for the data passing. I don't like the fact that any change of the format or adding types for overrides (e.g. the discussed IP) means a bit more changes. 
Is it fine to do it this way?

About the `serde_yaml`. We like the yaml as it's easy to work with and the configuration file is not expected to be large. So I left there the loading from yaml for now. If you consider it should really be the json, I can change it.